src/segy/file.py

```diff
@@ -196,35 +198,59 @@ def _parse_binary_header(self) -> None:
     @property
     def data(self) -> AbstractIndexer:
         """Way to access the file to fetch trace data only."""
+        transforms = [
+            TransformFactory.create("byte_swap", Endianness.LITTLE),
+        ]
+
+        if self.spec.trace.data_descriptor.format == ScalarType.IBM32:
+            transforms.append(TransformFactory.create("ibm_float", "to_ieee"))
+
         return DataIndexer(
             self.fs,
             self.url,
             self.spec.trace,
             self.num_traces,
             kind="data",
             settings=self.settings,
+            transforms=transforms,
         )
 
     @property
     def header(self) -> AbstractIndexer:
         """Way to access the file to fetch trace headers only."""
+        transforms = [
+            TransformFactory.create("byte_swap", Endianness.LITTLE),
+            TransformFactory.create("header_array"),
+        ]
```

Comment:
Does this also include handling floats in the trace headers?

Reply:
no that's a separate transform. if it needs to be done, can be added here (like the check for ibm32)